QRCoder/Base64QRCode.cs

```diff
@@ -43,25 +33,81 @@ public string GetGraphic(int pixelsPerModule, string darkColorHtmlHex, string li
 
         public string GetGraphic(int pixelsPerModule, Color darkColor, Color lightColor, bool drawQuietZones = true, ImageType imgType = ImageType.Png)
         {
-            var base64 = string.Empty;
-            using (Bitmap bmp = qr.GetGraphic(pixelsPerModule, darkColor, lightColor, drawQuietZones))
+            if (imgType == ImageType.Png)
             {
-                base64 = BitmapToBase64(bmp, imgType);
+                var pngCoder = new PngByteQRCode(QrCodeData);
+
+                byte[] pngData;
+                if (darkColor == Color.Black && lightColor == Color.White)
+                {
+                    pngData = pngCoder.GetGraphic(pixelsPerModule, drawQuietZones);
+                }
+                else
+                {
+                    byte[] darkColorBytes;
+                    byte[] lightColorBytes;
+                    if (darkColor.A != 255 || lightColor.A != 255)
+                    {
+                        darkColorBytes = new byte[] { darkColor.R, darkColor.G, darkColor.B, darkColor.A };
+                        lightColorBytes = new byte[] { lightColor.R, lightColor.G, lightColor.B, lightColor.A };
+                    }
+                    else
+                    {
+                        darkColorBytes = new byte[] { darkColor.R, darkColor.G, darkColor.B };
+                        lightColorBytes = new byte[] { lightColor.R, lightColor.G, lightColor.B };
+                    }
+                    pngData = pngCoder.GetGraphic(pixelsPerModule, darkColorBytes, lightColorBytes, drawQuietZones);
+                }
+
+                return Convert.ToBase64String(pngData, Base64FormattingOptions.None);
             }
-            return base64;
+
+#if NETFRAMEWORK || NETSTANDARD2_0 || NET5_0 || NET6_0_WINDOWS
+            if (
+#if NET6_0_OR_GREATER
```

Comment:
I was following the pattern in the rest of the program:
1. Up to .NET 5 contains a reference to System.Drawing.Common
2. .NET 6 contains the attributes to notate windows requirements

I'd probably do it different, but then we should change it throughout the library.

If I remember right, the `if` condition exists to eliminate a compiler warning.

Reply:
> .NET 6 contains the attributes to notate windows requirements
Yes, that's also what I would expect, but it looked to me as this wouldn't be the case here. 
You're using NET6_0_OR_GREATER to control if the attribute is set. The ...OR_GREATER flag (as per my understanding) means that .NET 6 or newer, regardless of the OS platform would be valid (would get the attribute). Correct?

But in practice this never would happen, because two lines earlier (line 65) the NET6_0_WINDOWS flag is used. So line 67 can only be reached on Windows platforms. 

Sorry if this sounds confusing - I sometimes get lost when it comes to all the target frameworks and their flags.

Reply:
NET6_0_OR_GREATER evaluates to true for the `net6.0-windows` tfm

Reply:
> The ...OR_GREATER flag (as per my understanding) means that .NET 6 or newer, regardless of the OS platform would be valid (would get the attribute). Correct?

Yes

> But in practice this never would happen, because two lines earlier (line 65) the NET6_0_WINDOWS flag is used. So line 67 can only be reached on Windows platforms.

It would also get hit for .NET 5 and prior.  But the attribute does not exist on .NET Framework, for example.  So the ifdef (or a polyfill) is required.

Reply:
But does the code work in .NET7 or .NET8? There neither NET5_0 nor NET6_0_WINDOWS should be set. 
So would it be a good idea to replace:
`#if NETFRAMEWORK || NETSTANDARD2_0 || NET5_0 || NET6_0_WINDOWS` 
with
`#if NETFRAMEWORK || NETSTANDARD2_0 || NET5_0_OR_GREATER`
?

Reply:
Since there is no target compilation target for net7 or net8, net6 is going to be used when .NET 7 is in use.  So ifdefs (which are based on the compilation target) are going to match on net6.  It's correct as-is.  But I'll make a subsequent PR with all the ifdefs cleaned up and we can review again.